packages/core/content-releases/server/src/bootstrap.ts

```diff
@@ -1,93 +1,44 @@
-                target_id: id,
-              },
-            });
-
-            // We update the status of each release after delete the actions
-            for (const release of releases) {
-              getService('release', { strapi }).updateReleaseStatus(release.id);
-            }
-          }
-        } catch (error) {
-          // If an error happens we don't want to block the delete entry flow, but we log the error
-          strapi.log.error('Error while deleting release actions after entry delete', { error });
-        }
-      },
       /**
-       * deleteMany hook doesn't return the deleted entries ids
-       * so we need to fetch them before deleting the entries to save the ids on our state
-       */
-      async beforeDeleteMany(event) {
-        const { model, params } = event;
-        // @ts-expect-error TODO: lifecycles types looks like are not 100% finished
-        if (model.kind === 'collectionType' && model.options?.draftAndPublish) {
-          const { where } = params;
-          const entriesToDelete = await strapi.db
-            .query(model.uid)
-            .findMany({ select: ['id'], where });
-          event.state.entriesToDelete = entriesToDelete;
-        }
-      },
-      /**
-       * We delete the release actions related to deleted entries
-       * We make this only after deleteMany is succesfully executed to avoid errors
+       * deleteMany is still used outside documents service, for example when deleting a locale
        */
       async afterDeleteMany(event) {
         try {
-          const { model, state } = event;
-          const entriesToDelete = state.entriesToDelete;
-          if (entriesToDelete) {
+          const model = strapi.getModel(event.model.uid as UID.Schema);
+          // @ts-expect-error TODO: lifecycles types looks like are not 100% finished
+          if (model.kind === 'collectionType' && model.options?.draftAndPublish) {
+            const { where } = event.params;
+
             const releases = await strapi.db.query(RELEASE_MODEL_UID).findMany({
               where: {
                 actions: {
-                  target_type: model.uid,
-                  target_id: {
-                    $in: (entriesToDelete as Array<{ id: Data.ID }>).map((entry) => entry.id),
-                  },
+                  contentType: model.uid,
+                  locale: where.locale ?? null,
+                  ...(where.documentId && { entryDocumentId: where.documentId }),
                 },
               },
             });
 
             await strapi.db.query(RELEASE_ACTION_MODEL_UID).deleteMany({
               where: {
-                target_type: model.uid,
-                target_id: {
-                  $in: (entriesToDelete as Array<{ id: Data.ID }>).map((entry) => entry.id),
-                },
+                contentType: model.uid,
+                locale: where.locale ?? null,
+                ...(where.documentId && { entryDocumentId: where.documentId }),
```

Comment:
could this create a recursive infinite loop? (deleteMany triggering this same lifecycle)

Reply:
It's a good point, honestly never thought about that. But you can see at line 15 that this subscriber is added only for D&P ContentTypes, and ReleaseAction doesn't has D&P, so the subscriber is not registered for it